experimental/benchmark/src/application/native.rs

```diff
@@ -0,0 +1,136 @@
+        match self.points_of_interest.entries.get(&request.get_ref().id) {
+            Some(point) => {
+                debug!("Found Point Of Interest: {:?}", point);
+                Ok(Response::new(GetPointOfInterestResponse {
+                    point_of_interest: Some(point.clone()),
+                }))
+            }
+            None => {
+                let err = tonic::Status::new(tonic::Code::Internal, "Empty database");
+                error!("{:?}", err);
+                Err(err)
+            }
+        }
+    }
+
+    async fn list_points_of_interest(
+        &self,
+        _request: Request<ListPointsOfInterestRequest>,
+    ) -> Result<Response<ListPointsOfInterestResponse>, Status> {
+        unimplemented!();
+    }
+}
+
+pub struct NativeApplication {
+    _handle: JoinHandle<()>,
+    client: TrustedDatabaseClient<tonic::transport::channel::Channel>,
+}
+
+impl NativeApplication {
+    pub async fn start(database: &Database, port: u16) -> Self {
+        info!("Running backend database");
+        let handle = tokio::spawn(NativeApplication::create_server(
+            database.points_of_interest.clone(),
+            port,
+        ));
+        let client = NativeApplication::create_client(port).await;
+        NativeApplication {
+            _handle: handle,
+            client,
+        }
+    }
+
+    async fn create_server(database: PointOfInterestMap, port: u16) {
+        let address = format!("[::]:{}", port)
+            .parse()
+            .expect("Couldn't parse address");
+        let handler = TrustedDatabaseService {
+            points_of_interest: database,
+        };
+        Server::builder()
+            .add_service(TrustedDatabaseServer::new(handler))
+            .serve(address)
+            .await
+            .expect("Couldn't start server");
+    }
+
+    async fn create_client(port: u16) -> TrustedDatabaseClient<tonic::transport::channel::Channel> {
+        let address = format!("https://localhost:{}", port)
+            .parse()
+            .expect("Couldn't parse address");
+        let channel = Channel::builder(address)
+            .connect()
+            .await
+            .expect("Couldn't connect to Oak Application");
+
+        let mut label = Vec::new();
+        Label::public_untrusted()
+            .encode(&mut label)
+            .expect("Error encoding label");
+        TrustedDatabaseClient::with_interceptor(channel, move |mut request: Request<()>| {
+            request.metadata_mut().insert_bin(
+                oak_abi::OAK_LABEL_GRPC_METADATA_KEY,
+                MetadataValue::from_bytes(label.as_ref()),
+            );
+            Ok(request)
+        })
+    }
+}
+
+#[async_trait]
+impl Application for NativeApplication {
+    async fn send_request(&mut self, id: &str) -> Result<(), tonic::Status> {
+        let request = Request::new(GetPointOfInterestRequest { id: id.to_string() });
+        self.client.get_point_of_interest(request).await?;
```

Comment:
Perhaps repeat the comment about discarding the request because irrelevant to the benchmark (though TBH it seems simpler to just make this method return the actual response to the caller, and just ignore it there)